### PR TITLE
test: Add basic functional tests for the database rules file methods

### PR DESCRIPTION
**Pull request by shepmaster**

### CODE REVIEW

File: iox_object_store/src/lib.rs [hunk 1]
```diff
@@ -503,15 +503,66 @@ mod tests {
         assert!(ctf.contains(&t2));
     }
 
-    #[test]
-    fn db_rules_should_be_a_file() {
+    fn make_db_rules_path(object_store: &ObjectStore, db_name: &str) -> Path {
```
Comment (carols10cents) [inline iox_object_store/src/lib.rs:506]: Why doesn't this take a server id?

**Review by carols10cents — COMMENTED** [review]
Just one little question!

File: iox_object_store/src/lib.rs (hunk 1 above)
Comment (shepmaster) [reply iox_object_store/src/lib.rs]: It does now!

File: iox_object_store/src/lib.rs [hunk 2]
```diff
@@ -503,15 +503,66 @@ mod tests {
...
+            .unwrap()
+            .next()
+            .await
+            .unwrap()
+            .unwrap();
+
+        assert_eq!(original_file_content, actual_content);
+
+        // GET
+        let updated_file_content = Bytes::from("goodbye moon");
```
Comment (alamb) [inline iox_object_store/src/lib.rs:541]: lol

**Review by alamb — APPROVED** [review]
Thanks @shepmaster

**Pull request merged**